hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/distribution/DistributionTrait.java

```diff
@@ -76,6 +76,32 @@ public boolean satisfies(RelTrait targetTrait) {
         return this.equals(targetTrait);
     }
 
+    /**
+     * Checks whtehr the result set of the node having this trait is guaranteed to exist on all members that will execute a
+     * fragment with this node.
+     *
+     * @return {@code true} if the full result set exists on all participants of the fragment hosting this node.
+     */
+    public boolean isFullResultSetOnAllParticipants() {
```

Comment:
It would be nice to have isolated unit tests for this method to ensure the stability of the implementation.

Reply:
Added in https://github.com/hazelcast/hazelcast/pull/18013/commits/653e813cdd461e412e028d233b01671151d33db3